src/classes/PS_CommitmentRequest.cls

```diff
@@ -122,28 +122,61 @@ public inherited sharing class PS_CommitmentRequest {
         set;
     }
 
+    /**
+    * @description Recurring Donation schedule service user to build schedules used in a commitment request
+    */
+    private RD2_ScheduleService scheduleService {
+        get {
+            if (scheduleService == null) {
+                scheduleService = new RD2_ScheduleService();
+            }
+            return scheduleService;
+        }
+        set;
+    }
+
     /***
-    * @description Creates an HttpRequest for a new commitment creation
+    * @description Creates an HttpRequest for a new commitment creation or
+    * existing commitment update depending on the prodived commitment Id
+    * @param commitmentId Elevate recurring commitment Id
     * @param jsonRequestBody JSON containing parameters for the purchase call request body
     * @return HttpRequest
     */
-    public HttpRequest buildCreateRequest(String jsonRequestBody) {
+    public HttpRequest buildRequest(String commitmentId, String jsonRequestBody) {
```

Comment:
The latest change, primarily in the [commit](https://github.com/SalesforceFoundation/NPSP/pull/6193/commits/a3620f1cb257fe732f1813f6b0e108e76ab3e4cf), has been done to issue an edit request instead of bypassing the commitment API request when commitment is created, and the RD was changed  after erroring out first (see AC7).
This way we are building a path for the edit that will work with create instead of patching the code to work for AC7 but later on changing it when edit changes are introduced or when user changes the Amount or schedule after fixing an error for RD insert failure for which commitment has been created. Please let me know if there is any issue with this approach.
@rponti-sforg , the PR can be retested (AC7 and happy path ).

Reply:
@RadSFDeveloper Re-testing.  Will update you with any questions or concerns.